### PR TITLE
feat: NotificationChannel and alert window type for screencast oreo s…

### PR DESCRIPTION
**Pull request by jasonchuang**
screencast oreo support

### CODE REVIEW

File: Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/OverlayLayout.java [hunk 1]
```diff
@@ -21,15 +23,30 @@
     protected final Listener mListener;
 
     protected WindowManager.LayoutParams createLayoutParams() {
+        int alertWindowType = Utils.supportAndroidOreo() ?
+                WindowManager.LayoutParams.TYPE_APPLICATION_OVERLAY :
+                WindowManager.LayoutParams.TYPE_SYSTEM_ERROR;
         final WindowManager.LayoutParams params = new WindowManager.LayoutParams(WRAP_CONTENT, WRAP_CONTENT,
```
Comment (jasonchuang) [inline Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/OverlayLayout.java]: TYPE_SYSTEM_ERROR 跟 TYPE_APPLICATION_OVERLAY 主要差別在於 是不是位於 view hierarchy 的最上層, e.x. 8.0 之前, overlay 可以蓋在 status bar or navigation bar 上面移動, 8.0 之後就會被壓在下面, user 點不到

File: Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/MyScreencastSession.java [hunk 2]
```diff
@@ -318,16 +324,33 @@ public void onSuccess(Void aVoid) {
...
+    @Override
+    public NotificationChannel getNotificationChannel() {
+        if (!Utils.supportAndroidOreo()) {
+            return null;
+        }
+
+        NotificationChannel channel = new NotificationChannel(
+                mContext.getString(R.string.screencast_notification_channel_id),
+                mContext.getString(R.string.screencast_notification_channel_name),
+                IMPORTANCE_LOW);
```
Comment (ItsRay) [inline Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/MyScreencastSession.java]: 是不是要 high

File: Streaming/src/main/java/io/straas/android/sdk/streaming/demo/Utils.java [hunk 3]
```diff
@@ -41,4 +42,8 @@ private static String formatBitrate(Context context, int number) {
         return context.getResources().getString(R.string.bitrate_suffix, value,
                 context.getString(suffix));
     }
+
+    public static boolean supportAndroidOreo() {
```
Comment (ItsRay) [inline Streaming/src/main/java/io/straas/android/sdk/streaming/demo/Utils.java]: 這個看覺得會不會怪怪的, laterThanOreo ？或者維持

File: Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/OverlayLayout.java [hunk 4]
```diff
@@ -21,15 +23,30 @@
...
                 WindowManager.LayoutParams.FLAG_LAYOUT_INSET_DECOR | WindowManager.LayoutParams.FLAG_LAYOUT_IN_SCREEN,
                 PixelFormat.TRANSLUCENT);
         params.gravity = getLayoutGravity();
+        params.y = getVerticalMargin(params.gravity);
         return params;
     }
 
+    private int getVerticalMargin(int gravity) {
+        // top gravity then align to status bar height
+        if ((gravity & Gravity.TOP) == Gravity.TOP) {
```
Comment (ItsRay) [inline Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/OverlayLayout.java]: 這個可以簡化成 `gravity == Gravity.TOP` 嗎？

Comment (james12695) [reply Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/OverlayLayout.java]: 沒辦法, gravity 可能會是 LEFT | TOP

File: Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/MyScreencastSession.java (hunk 2 above)
Comment (james12695) [reply Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/MyScreencastSession.java]: discuss

Comment (jasonchuang) [reply Streaming/src/main/java/io/straas/android/sdk/streaming/demo/screencast/MyScreencastSession.java]: IMPORTANCE_DEFAULTD
default notification importance: shows everywhere, `makes noise, but does not visually intrude`
IMPORTANCE_HIGH
Higher notification importance: shows everywhere, `makes noise and peeks`
IMPORTANCE_LOW
Low notification importance: shows everywhere, but `is not intrusive`

**Pull request merged**